File: docs/site/samples/python/examples/guides_guard_construct.py
"""The one-variable `ask` prompt used throughout the guard guide: `topic` is
declared untrusted (trusted: false). Standalone — run directly or under pytest."""

from pathlib import Path

from prompting_press import Prompt
from pydantic import BaseModel, Field

# The prompt: `topic` is declared untrusted (trusted: false).
# The caller reads the definition; the library does no file I/O itself.
# Resolve the file next to this program (a real app uses its own path).
_HERE = Path(__file__).parent
ask = Prompt.from_yaml((_HERE / "ask.yaml").read_text())


# The typed vars model handed to render().
class Ask(BaseModel):
    topic: str = Field(min_length=1)


def main() -> None:
    # Without the guard the body renders plainly.
    result = ask.render(Ask, data={"topic": "rivers"})
    assert result.text == "Tell me about rivers."
    assert result.guard is None
    print(result.text)


if __name__ == "__main__":
    main()

File: docs/site/samples/typescript/examples/guides_guard_construct.ts
// The one-variable `ask` prompt used throughout the guard guide: `topic` is
// declared untrusted (trusted: false). Standalone — run under `node --test`.

import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { test } from "node:test";
import { Prompt } from "prompting-press";
import { z } from "zod";

// The caller reads the definition; the library does no file I/O itself.
// Resolve the file next to this program (a real app uses its own path).
const defFile = (name: string) => fileURLToPath(new URL(name, import.meta.url));

// The prompt: `topic` is declared untrusted (trusted: false).
const ask = Prompt.fromYaml(readFileSync(defFile("ask.yaml"), "utf8"));

// The typed vars schema handed to render().
const Ask = z.object({ topic: z.string().min(1) });

test("the ask prompt renders plainly without the guard", () => {
	const result = ask.render(Ask, { topic: "rivers" });
	assert.equal(result.text, "Tell me about rivers.");
	assert.equal(result.guard, null);
});
